coalib/testing/LocalBearTestHelper.py

```diff
@@ -193,13 +171,12 @@ def check_invalidity(self,
         :param create_tempfile:  Whether to save lines in tempfile if needed.
         :param tempfile_kwargs:  Kwargs passed to tempfile.mkstemp().
         """
+        assert isinstance(self, BaseTestHelper)
         assert isinstance(self, unittest.TestCase)
-        self.assertIsInstance(local_bear,
-                              LocalBear,
-                              msg='The given bear is not a local bear.')
-        self.assertIsInstance(lines,
-                              (list, tuple),
-                              msg='The given lines are not a list.')
+        assert isinstance(
+            local_bear, LocalBear), 'The given bear is not a local bear.'
+        assert isinstance(
+            lines, (list, tuple)), 'The given lines are not a list.'
```

Comment:
Due to pytest's assertion introspection we already get [very good failure reports](https://docs.pytest.org/en/latest/example/reportingdemo.html#tbreportdemo). This eliminates the need for custom messages in these assert statements. 

I'm not entirely sure, if that feature is enabled for unittest.TestCases. That's something you'll need to verify first.

Reply:
changing these to using `assert ...` is not part of "Inherit from ..." if this class also inherits from `unittest.TestCase`.

So you can investigate whether these are needed later.
Also rather than having custom asserts here, we can and should use arg annotations and our decororator which enforces those.  Also noted this in https://github.com/coala/coala/pull/5522#discussion_r195143624
But again, changes to these assertions is a functional change which doesnt need to be in the same commit as "Inherit from ...", and it does need a separate issue created where you *show* that the output can be improved.  As Niklas has said, pytest already gives very detailed information about any failures, and pytest does that even for `unittest.TestCase` subclasses, so any custom type error messages need to be better than pytest's output.

Reply:
@NiklasMM, when I removed the custom message, `unittest`'s `self.assertIsInstance(..)` gave 
```py
E   AssertionError: '' is not an instance of (<class 'list'>, <class 'tuple'>)
```
 
but `pytest`'s `assert` gave
```py
E       AssertionError
```